Cygwin job is broken on "Set up virtualenv" step

The Cygwin CI test job is failing now, including [when rerun](https://github.com/EliahKagan/GitPython/actions/runs/13041816861/job/37557892745) at the tip of the main branch where it formerly passed. The failure occurs in the "Set up virtualenv" step, which runs `python -m venv .venv`. The log does not how a detailed error message, just:

```text
Run python -m venv .venv
  python -m venv .venv
  echo 'BASH_ENV=.venv/bin/activate' >>"$GITHUB_ENV"
  shell: C:\tools\cygwin\bin\bash.exe --login --norc -eo pipefail -o igncr "{0}"
  env:
    CHERE_INVOKING: 1
    CYGWIN_NOWINPATH: 1
    CYGWIN: 
+ python -m venv .venv
Error: Command '['/cygdrive/d/a/GitPython/GitPython/.venv/bin/python', '-Im', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.
Error: Process completed with exit code 1.
```

I first noticed this when checking the current status of #1988 (see https://github.com/gitpython-developers/GitPython/pull/1988#issuecomment-2672396018), but it is not specific to that PR.

I do not know why this happens.

--- a/test/test_installation.py
+++ b/test/test_installation.py
@@ -4,11 +4,19 @@
 import ast
 import os
 import subprocess
+import sys
+
+import pytest
 
 from test.lib import TestBase, VirtualEnvironment, with_rw_directory
 
 
 class TestInstallation(TestBase):
+    @pytest.mark.xfail(
+        sys.platform == "cygwin" and "CI" in os.environ,
+        reason="Trouble with pip on Cygwin CI, see issue #2004",
+        raises=subprocess.CalledProcessError,
+    )
     @with_rw_directory
     def test_installation(self, rw_dir):
         venv = self._set_up_venv(rw_dir)

